blackbox/docs/general/dql/joins.txt

```diff
@@ -201,17 +201,31 @@ Example with ``within`` scalar function::
     +--------------+---------+
     SELECT 2 rows in set (... sec)
 
-However, there are no specific optimisations for certain join conditions such
-as ``=`` (equals) yet. The JOIN operation is implemented as a generic nested
-loop that invokes the operation on every record of the left table with every
-record of the right table. However, these filters are applied in a distributed
-way for each shard of the used tables.
+.. NOTE::
+
+    The JOIN operation is implemented as a generic nested loop that evaluates
+    the join conditions on every record of the left table with every record of
+    the right table in a distributed manner (for each shard of the used
+    tables).
+
+    However, performance for `Equi-Joins`_  can be substantially improved by
+    using the `Hash Join`_ algorithm. Hash Join is currently an experimental
+    feature which can be enabled on demand with the session setting
+    `enable_hashjoin`::
+
+      SET enable_hashjoin=true
```

Comment:
I'd be in favor of keeping all of this undocumented until we've some actual data on the runtime behaviour and until we've a better picture on how we're going to integrate this in the planner. 

Otherwise we'd have to go into more detail here regarding the memory requirements and the possibility of failure if we don't implement some sort of failsafe or fallback.

Reply:
I get your points there, but on the other hand if we release a version without the automated Planner decision then would be weird to include the feature in a blog post and have not even a slight mention of that in our docs. And I believe that this feature is something we want to "advertise". Any more thoughts?